aggregator/src/aggregator.rs

```diff
@@ -188,6 +188,13 @@ impl AggregatorMetrics {
     }
 }
 
+/// Represents the result of a Job Continue operation, which varies whether that was synchronous
+/// or async.
+enum AggregationJobContinueResult {
+    Sync { resp: AggregationJobResp },
+    Async { step: AggregationJobStep },
```

Comment:
nit: I'm fine with this as-is, but in a case like this where each enum variant only contains a single value, I think using the struct notation is a bit too verbose. In tuple notation you'd have just:
```suggestion
    Sync(AggregationJobResp),
    Async(AggregationJobStep),
```

And it's a bit more compact when instantiating the values too.